app/javascript/lib/validation/validate-file-content.js

```diff
@@ -263,12 +263,16 @@ export async function parseMetadataFile(chunker, mimeType, fileOptions) {
   // add other header validations here
 
   const dataObj = {} // object to track multi-line validation concerns
-  await chunker.iterateLines((rawline, lineNum, isLastLine) => {
-    const line = parseLine(rawline, delimiter)
-    issues = issues.concat(validateUniqueCellNamesWithinFile(line, isLastLine, dataObj))
-    issues = issues.concat(validateMetadataLabelMatches(headers, line, isLastLine, dataObj))
-    issues = issues.concat(validateGroupColumnCounts(headers, line, isLastLine, dataObj))
+  await chunker.iterateLines({
+    func: (rawline, lineNum, isLastLine) => {
+      issues = issues.concat(timeOutCSFV(chunker))
```

Comment:
```suggestion
      if (lineNum % 100 === 0) {
        issues = issues.concat(timeOutCSFV(chunker))
      }
```

Reply:
Yeah, I think you're right that this is an argument to move the time check to inside the chunker, but then that will require some API changes to chunker (chunker would raise a "TimeoutException" or somesuch), which are outside the scope of the PR

Yeah, definitely about tradeoffs, is a x% speedup worth a y% chance that it will end up taking longer on long runs because the validation read too many lines?